File: python/resistor-color-duo/resistor_color_duo.py
def value(colors):
    resistor_codes = {
        "black": 0,
        "brown": 1,
        "red": 2,
        "orange": 3,
        "yellow": 4,
        "green": 5,
        "blue": 6,
        "violet": 7,
        "grey": 8,
        "white": 9,
    }

    color_values = []

    for color in colors:
        color_values.append(str(resistor_codes.get(color)))

    return int(''.join(color_values[:2]))

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;


class ResistorColorDuo {
    int value(String[] colors) {
        String resistorCode = String.format("%s%s", colorCode(colors[0]), colorCode(colors[1]));
        return Integer.parseInt(resistorCode);
    }

    int colorCode(String color) {
        return Arrays.asList(colors()).indexOf(color);
    }

    String[] colors() {
        return new String[] {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
    }
}
